conductr_cli/resolvers/uri_resolver.py

```diff
@@ -37,10 +38,31 @@ def resolve_file(cache_dir, uri, auth=None):
         return False, None, None
 
 
-def load_bundle_from_cache(cache_dir, uri):
+def load_bundle_from_cache(cache_dir, uri, offline_mode=False):
+    """
+    Tries to load a bundle from the cache directory.
+    If offline mode is enabled and the given uri equals a bundle name without slashes, e.g. 'visualizer'
+    then it tries to resolve the last modified bundle from the cache directory by the given uri.
+    Otherwise, when the supplied uri is a local filesystem, the file is not loaded from cache so that the
+    local file can be loaded directly.
+    :param cache_dir: the cache directory
+    :param uri: the bundle uri. Can be either a bundle name, e.g. 'visualizer, an http or file uri
+    :param offline_mode: the offline mode flag
+    :return: a tuple of (is_cached, bundle_name, bundle_uri)
+    """
     # When the supplied uri is a local filesystem, don't load from cache so file can be used as is
     parsed = urlparse(uri, scheme='file')
-    if parsed.scheme == 'file':
+    if offline_mode and is_bundle_name(uri):
```

Comment:
Would you consider having a separate resolver called `offline_resolver`?

Having a separate `offline_resolver` means the offline logic can be contained within the `offline_resolver`, and the `uri_resolver` implementation will be purely uri-related.

I think this would be a cleaner approach - wdyt?

If you do agree with this approach, I think optionally the `offline_mode` input argument can be removed from the `load_bundle_from_cache ` method.